File: algorithms/2_medium/S/02679_sum_in_a_matrix/sort.c
static inline int max_int(const int a, const int b)
{
    return a > b ? a : b;
}

static int cmp(const void* a_ptr, const void* b_ptr)
{
    const int a = *(int*)a_ptr;
    const int b = *(int*)b_ptr;
    return (a > b) - (a < b);
}

int matrixSum(int** nums, const int rows, const int* cols_data)
{
    assert(rows > 0 && nums != NULL && cols_data != NULL && cols_data[0] > 0);
    const int cols = cols_data[0];
    int score = 0;

    for (int i = 0; i < rows; i += 1) {
        qsort(nums[i], (size_t)cols, sizeof(**nums), cmp);
    }

    for (int j = 0; j < cols; j += 1) {
        int max = INT_MIN;

        for (int i = 0; i < rows; i += 1) {
            max = max_int(max, nums[i][j]);
        }

        score += max;
    }

    return score;
}

File: algorithms/2_medium/S/02679_sum_in_a_matrix/sort.rs
impl Solution {
    pub fn matrix_sum(mut nums: Vec<Vec<i32>>) -> i32 {
        let rows = nums.len();
        let cols = nums[0].len();
        let mut score = 0_i32;

        for i in 0..rows {
            nums[i].sort_unstable();
        }

        for j in 0..cols {
            let mut max = i32::MIN;

            for i in 0..rows {
                max = max.max(nums[i][j]);
            }

            score += max;
        }

        score
    }
}
